### PR TITLE
Add missing versions of VectorTools::interpolate_boundary_values for hp::MappingCollection

### PR DESCRIPTION
**Pull request by peterrum**

### CODE REVIEW

**Review by kronbichler — APPROVED** [review]

File: include/deal.II/numerics/vector_tools_boundary.templates.h [hunk 1]
```diff
@@ -440,6 +440,25 @@ namespace VectorTools
...
+    const hp::MappingCollection<dim, spacedim> &mapping,
+    const DoFHandler<dim, spacedim> &           dof,
+    const types::boundary_id                    boundary_component,
+    const Function<spacedim, number> &          boundary_function,
+    std::map<types::global_dof_index, number> & boundary_values,
+    const ComponentMask &                       component_mask)
+  {
+    std::map<types::boundary_id, const Function<spacedim, number> *>
+      function_map;
+    function_map[boundary_component] = &boundary_function;
```
Comment (bangerth) [inline include/deal.II/numerics/vector_tools_boundary.templates.h:455]: You can do this more easily if you say
```
  std::map<...,...> function_map = {{boundary_component, &boundary_function}};
```

File: include/deal.II/numerics/vector_tools_boundary.templates.h [hunk 2]
```diff
@@ -530,6 +549,54 @@ namespace VectorTools
...
+      &                        function_map,
+    AffineConstraints<number> &constraints,
+    const ComponentMask &      component_mask_)
+  {
+    std::map<types::global_dof_index, number> boundary_values;
+    interpolate_boundary_values(
+      mapping, dof, function_map, boundary_values, component_mask_);
+    typename std::map<types::global_dof_index, number>::const_iterator
+      boundary_value = boundary_values.begin();
+    for (; boundary_value != boundary_values.end(); ++boundary_value)
```
Comment (bangerth) [inline include/deal.II/numerics/vector_tools_boundary.templates.h:567]: Just use a range-based for loop :-)

File: include/deal.II/numerics/vector_tools_boundary.templates.h [hunk 3]
```diff
@@ -530,6 +549,54 @@ namespace VectorTools
...
+    const hp::MappingCollection<dim, spacedim> &mapping,
+    const DoFHandler<dim, spacedim> &           dof,
+    const types::boundary_id                    boundary_component,
+    const Function<spacedim, number> &          boundary_function,
+    AffineConstraints<number> &                 constraints,
+    const ComponentMask &                       component_mask)
+  {
+    std::map<types::boundary_id, const Function<spacedim, number> *>
+      function_map;
+    function_map[boundary_component] = &boundary_function;
```
Comment (bangerth) [inline include/deal.II/numerics/vector_tools_boundary.templates.h:593]: same as above

**Review by bangerth — COMMENTED** [review]
Almost -- just use a few C++11 features :-)

**Comment by bangerth** [comment]
Ah, I didn't realize that @kronbichler had already approved it. Either way is fine with me.

**Pull request merged**